### PR TITLE
Convert `conda.core.initialize` unittest to pytest

### PR DESCRIPTION
**Pull request by kenodegard**
### Description

Flatten unittest class into pytest functions and refacted setup/teardown as fixture.

Also contains minimal reformatting of c-style string formatting.

Prework for adding new tests in #11865.

### Checklist - did you ...

- [ ] ~Add a file to the `news` directory ([using the template](../blob/main/news/TEMPLATE)) for the next release's release notes?~
     
- [x] Add / update necessary tests?
- [ ] ~Add / update outdated documentation?~

### CODE REVIEW

File: tests/core/test_initialize.py [hunk 1]
```diff
@@ -19,709 +17,767 @@
...
-    from unittest.mock import Mock, patch
-except ImportError:
-    from unittest.mock import Mock, patch
-
 
 log = getLogger(__name__)
 
 
-class InitializeTests(TestCase):
+@pytest.fixture
```
Comment (dholth) [inline tests/core/test_initialize.py:46]: Try adding `fixture(scope="module")` to run this code once, instead of once per test.

**Review by dholth — APPROVED** [review]

File: tests/core/test_initialize.py (hunk 1 above)
Comment (kenodegard) [reply tests/core/test_initialize.py]: I wasn't sure if we needed the context to be reset once or per function, it looks like everything is good when I change it

Comment (dholth) [reply tests/core/test_initialize.py]: I checked the docs, and setUp and tearDown are run after each method. Sorry for the inconvenience.

It does look like some of the test alter context, and would benefit from the `reset_context()` call.

https://docs.python.org/3/library/unittest.html#unittest.TestCase.setUp

File: tests/core/test_initialize.py [hunk 2]
```diff
@@ -19,964 +18,1065 @@
...
 from conda.models.enums import LinkType
 from conda.testing.helpers import tempdir
 
-try:
-    from unittest.mock import Mock, patch
-except ImportError:
-    from unittest.mock import Mock, patch
-
 
 log = getLogger(__name__)
```
Comment (sven6002) [inline tests/core/test_initialize.py:43]: Could be removed..

**Review by sven6002 — APPROVED** [review]
Refactoring is a possible ticket for future.. but it not deterring the working of the test cases, in test_initialize.py

File: tests/core/test_initialize.py [hunk 3]
```diff
@@ -1,14 +1,12 @@
...
-
-
 from distutils.sysconfig import get_python_lib
 from logging import getLogger
 import ntpath
 import os
 from os.path import abspath, dirname, isfile, join, realpath
 import sys
-from unittest import TestCase
+from unittest.mock import patch
```
Comment (dholth) [inline tests/core/test_initialize.py:9]: there is a pytest fixture called `mocker`

Comment (dholth) [reply tests/core/test_initialize.py:9]: It is the unittest.mock interface but it cleans up automatically. Example https://github.com/conda/conda-package-handling/blob/conda-package-handling-2.0/tests/test_cli.py#L30

**Review by dholth — APPROVED** [review]
LGTM

**Pull request merged**